File: uva_11029.py
import math
def mod(a, n):
    N=1000
    ans=a if n%2 else 1
    while int(n/2):
        n=int(n/2)
        temp=(a*a)%N
        if n%2:
            ans=(ans*temp)%N
        a=temp
    return ans
def first3digit(n, k):
    f=k*math.log10( n)-int(k*math.log10(n))
    return int(pow(10, 2+f))

N=int(input())
for _ in range(N):
    data=input().split()
    data[0], data[1]=int(data[0]), int(data[1])
    last3digit=mod(data[0]%1000, data[1])
    last3digit="%03d"%last3digit
    print("{}...{}".format(first3digit(data[0], data[1]),\
                            last3digit))

File: uva_11029.cpp
# include <cmath>
# include <iomanip>
# include <iostream>
using namespace std;

int mod(int n, int k){
    int N=1000, ans, temp;
    if(k%2)
        ans=n;
    else
        ans=1;
    while(k/2){
        k/=2;
        temp=(n*n)%N;
        if (k%2)
            ans=(ans*temp)%N;
        n=temp;
    }
    return ans;
}
int first3digit(int n, int k){
    double f=k*log10(n*1.0)-int(k*log10(n*1.0));
    return (int)pow(10, 2+f);;
}
int main(void){
    int N, n, k;
    cin >> N;
    while(N--){
        cin >> n >> k;
        cout<<first3digit(n, k)<<"..."
        <<setw(3)<<setfill('0')<< mod(n%1000, k)<<endl;
    }
    return 0;
}